plenum/server/request_managers/write_request_manager.py

```diff
@@ -348,13 +322,23 @@ def do_taa_validation(self, request: Request, req_pp_time: int, config):
                 .format(ledger_id)
             )
 
-        r_taa_a_digest = request.taaAcceptance[f.TAA_ACCEPTANCE_DIGEST.nm]
-        if r_taa_a_digest != taa_digest:
+        taa = None
+        r_taa_a_digest = request.taaAcceptance.get(f.TAA_ACCEPTANCE_DIGEST.nm)
+        taa_data = self.get_taa_data(digest=r_taa_a_digest)
+        if taa_data is not None:
+            (taa, taa_seq_no, taa_txn_time), taa_digest = taa_data
+        else:
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "Incorrect Txn Author Agreement(digest={}) in the request".format(r_taa_a_digest)
```

Comment:
I think it's better to say:
`Accepted Txn Author Agreement (digest={}) is not found on the ledger. Please make sure that an existing Agreement is accepted.".format(r_taa_a_digest)`